Modules/Challenges/src/Challenges.Infrastructure/Persistence/Database/Repository/ChallengesRepository.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Backbone.Modules.Challenges.Application.Extensions;
+using Backbone.Modules.Challenges.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Challenges.Domain.Entities;
+using Backbone.Modules.Challenges.Domain.Ids;
+using Microsoft.EntityFrameworkCore;
+
+namespace Backbone.Modules.Challenges.Infrastructure.Persistence.Database.Repository;
+public class ChallengesRepository : IChallengesRepository
+{
+    private readonly DbSet<Challenge> _challenges;
+    private readonly ChallengesDbContext _dbContext;
+    
+    public ChallengesRepository(ChallengesDbContext dbContext)
+    {
+        _challenges = dbContext.Challenges;
+        _dbContext = dbContext;
+    }
+    public async Task<Challenge> Find(ChallengeId id, CancellationToken cancellationToken)
+    {
+        return await _challenges
+            .NotExpired()
+            .FirstWithId(id, cancellationToken);
+    }
+
+    public async Task<Challenge> Add(Challenge challenge, CancellationToken cancellationToken)
```

Comment:
Don't return the Challenge. This is not needed.

(I know you copied that from the `MessagesRepository`, but it was wrong there as well and will be fixed with the next PR)